File: AdventOfCode/2015/3a.pl
use strict;
use warnings;
use autodie;
use v5.10; # for say


open (my $file, '<', '3.inp');
chomp(my $inp = <$file>);
my %seen;
my $x = 0;
my $y = 0;
$seen{"0:0"} = 1; # Initial location is included
for my $ch (split '', $inp) {
    if ($ch eq '^') {
        $x++;
    } elsif ($ch eq '>') {
        $y++;
    } elsif ($ch eq '<') {
        $y--;
    } elsif ($ch eq 'v') {
        $x--;
    } else {
        die "Invalid Char";
    }
    $seen{"$x:$y"} = 1; # We could've used a 2D hash $seen{$x}{$y} but this works and easier to count keys.
}
say scalar keys %seen;

File: AdventOfCode/2016/3a.rb
count = 0
File.open('3.inp', 'r') do |f|
  f.each_line do |line|
    x, y, z = line.chomp.split(' ').map(&:to_i)
    next if x + y <= z
    next if x + z <= y
    next if y + z <= x
    count += 1
  end
end
p count
